src/core/toga/widgets/detailedlist.py

```diff
@@ -2,55 +2,108 @@
 
 
 class DetailedList(Widget):
-    """
+    """ A widget to hold data in a list form. Rows are selectable and can be deleted.
+    A updated function can be invoked by pulling the list down.
 
     Args:
-        id (str):
-        data:
-        on_delete:
-        on_refresh:
-        style:
+        id (str): An identifier for this widget.
+        data (list of `str`): List of strings which to display on the widget.
+        on_delete (``callable``): Function that is invoked on row deletion.
+        on_refresh (``callable``): Function that is invoked on user initialised refresh.
+        on_select (``callable``): Function that is invoked on row selection.
+        style (:class:`colosseum.CSSNode`): An optional style object. If no style is provided then
+            a new one will be created for the widget.
         factory (:obj:`module`): A python module that is capable to return a
             implementation of this class with the same name. (optional & normally not needed)
 
-    Todo:
-        * Add missing docstrings.
+    Examples:
+        >>> import toga
+        >>> def selection_handler(widget, selection):
+        >>>     print('Row {} of widget {} was selected.'.format(selection, widget))
+        >>>
+        >>> dlist = toga.DetailedList(data=['Item 0', 'Item 1', 'Item 2'], on_select=selection_handler)
     """
 
-    def __init__(self, id=None, data=None, on_delete=None, on_refresh=None, style=None, factory=None):
+    def __init__(self, id=None, data=None, on_delete=None, on_refresh=None, on_select=None, style=None, factory=None):
         super().__init__(id=id, style=style, factory=factory)
 
         self._impl = self.factory.DetailedList(interface=self)
 
         self.data = data
         self.on_delete = on_delete
         self.on_refresh = on_refresh
+        self.on_select = on_select
 
     def add(self, item):
+        """ Add a item to the list.
+
+        Args:
+            item (str): String add to the list.
+        """
         self._data.append(item)
         self._impl.add(item)
 
     @property
-    def data(self):
+    def data(self) -> list:
+        """ The data displayed in the rows of the list.
+
+        Returns:
+            The row data in form of a list.
+        """
         return self._data
 
     @data.setter
-    def data(self, data_list):
+    def data(self, data_list: list):
```

Comment:
This should be a data source, not just a list. (c.f., the way tree/table handles data). Ideally, we want to allow end users to have dynamic sources of data, and have them displayed as a detailed list.